commit/src/pcs.rs

```diff
@@ -28,96 +31,60 @@ pub trait Pcs<Val: Field, In: MatrixRows<Val>> {
-pub type OpenedValuesForPoint<F> = Vec<F>;
-
-pub trait UnivariatePcs<Val, EF, In, Challenger>: Pcs<Val, In>
-where
-    Val: Field,
-    EF: ExtensionField<Val>,
-    In: MatrixRows<Val>,
-    Challenger: FieldChallenger<Val>,
-{
-    fn open_multi_batches(
-        &self,
-        prover_data_and_points: &[(&Self::ProverData, &[Vec<EF>])],
-        challenger: &mut Challenger,
-    ) -> (OpenedValues<EF>, Self::Proof);
-
-    fn verify_multi_batches(
-        &self,
-        commits_and_points: &[(Self::Commitment, &[Vec<EF>])],
-        dims: &[Vec<Dimensions>],
-        values: OpenedValues<EF>,
-        proof: &Self::Proof,
-        challenger: &mut Challenger,
-    ) -> Result<(), Self::Error>;
-}
-
-/// A `UnivariatePcs` where the commitment process involves computing a low-degree extension (LDE)
-/// of each polynomial. These LDEs can be reused in other prover work.
-pub trait UnivariatePcsWithLde<Val, EF, In, Challenger>:
-    UnivariatePcs<Val, EF, In, Challenger>
-where
-    Val: Field,
-    EF: ExtensionField<Val>,
-    In: MatrixRows<Val>,
-    Challenger: FieldChallenger<Val>,
-{
-    type Lde<'a>: MatrixRows<Val> + MatrixGet<Val> + Sync
-    where
-        Self: 'a;
-
-    fn coset_shift(&self) -> Val;
-
-    fn log_blowup(&self) -> usize;
-
-    fn get_ldes<'a, 'b>(&'a self, prover_data: &'b Self::ProverData) -> Vec<Self::Lde<'b>>
-    where
-        'a: 'b;
+    fn natural_domain_for_degree(&self, degree: usize) -> Self::Domain;
 
-    // Commit to polys that are already defined over a coset.
-    fn commit_shifted_batches(
+    #[allow(clippy::type_complexity)]
+    fn commit(
         &self,
-        polynomials: Vec<In>,
-        coset_shift: &[Val],
+        evaluations: Vec<(Self::Domain, RowMajorMatrix<Val<Self::Domain>>)>,
     ) -> (Self::Commitment, Self::ProverData);
 
-    fn commit_shifted_batch(
+    fn get_evaluations_on_domain(
         &self,
-        polynomials: In,
-        coset_shift: Val,
-    ) -> (Self::Commitment, Self::ProverData) {
-        self.commit_shifted_batches(vec![polynomials], &[coset_shift])
-    }
-}
+        prover_data: &Self::ProverData,
+        idx: usize,
+        domain: Self::Domain,
+    ) -> RowMajorMatrix<Val<Self::Domain>>;
 
-pub trait MultivariatePcs<Val, EF, In, Challenger>: Pcs<Val, In>
-where
-    Val: Field,
-    EF: ExtensionField<Val>,
-    In: MatrixRows<Val>,
-    Challenger: FieldChallenger<Val>,
-{
-    fn open_multi_batches(
+    fn open(
         &self,
-        prover_data_and_points: &[(&Self::ProverData, &[Vec<EF>])],
+        // For each round,
+        rounds: Vec<(
+            &Self::ProverData,
+            // for each matrix,
+            Vec<
+                // points to open
+                Vec<Challenge>,
+            >,
+        )>,
         challenger: &mut Challenger,
-    ) -> (OpenedValues<EF>, Self::Proof);
+    ) -> (OpenedValues<Challenge>, Self::Proof);
 
-    fn verify_multi_batches(
+    #[allow(clippy::type_complexity)]
```

Comment:
hmm

Reply:
It's perfect 😊 (at least I don't see an obvious way to improve it)